pkg/apis/serving/v1beta1/inference_service_status.go

```diff
@@ -105,6 +111,96 @@ const (
 	IngressReady apis.ConditionType = "IngressReady"
 )
 
+type ModelStatus struct {
+	// Whether the available predictor endpoint reflects the current Spec or is in transition
+	// +kubebuilder:default=UpToDate
+	TransitionStatus TransitionStatus `json:"transitionStatus"`
```

Comment:
@Suresh-Nakkeran I think for single model `TransitionStatus` can be set to following state:
- `UpToDate` when `latestCreatedRevision` == `lastestReadyRevision`
- `InProgress` when `latestCreatedRevision` is still in progress
- `BlockedByFailedLoad` when `latestCreatedRevision` fails